cmd/soroban-cli/src/config/self_outdated_check.rs

```diff
@@ -0,0 +1,82 @@
+use crate::config::locator;
+use jsonrpsee_core::Serialize;
+use serde::Deserialize;
+use std::fs;
+
+const FILE_NAME: &str = "self_outdated_check.toml";
+
+/// The `SelfOutdatedCheck` struct represents the state of the self-outdated check.
+/// This state is global and stored in the `self_outdated_check.toml` file in
+/// the global configuration directory.
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct SelfOutdatedCheck {
+    /// The timestamp of the latest check for a new version of the CLI.
+    pub latest_check_time: u64,
+    /// The latest stable version of the CLI available on crates.io.
+    pub max_stable_version: String,
+    /// The latest version of the CLI available on crates.io, including pre-releases.
+    pub max_version: String,
```

Comment:
Could also use `Version` here. This will prevent needing to parse every time when it's used.

Reply:
Changed.